content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+- We would also like it to be as responsive as possible since it makes for a better user experience.
+
+Note that we don't really need brazing fast graphic drawing for this since the game will not render thousands of objects at once, so using WebGL is probably overkill. However, it's a good way to get into the topic and so we'll take the chance to do it that way. WebGL is fairly complicated to work with from scratch, so after doing a bit of research I have decided we can use PixiJS to render stuff a pretty simple set of Javascript libraries for a 2D renderer that runs over WebGL (which in turn means we are using hardware acceleration to render stuff). We can also go a step further and try to use it with Typescript, to get that sweet, sweet type-checking.
+
+## Starting out
+
+In the world of web development, it's common
+We are going to use a starter template, which will solve some of these issues for us. Specifically, the starter is the following: https://github.com/llorenspujol/parcel-pixijs-quickstarter. It includes Parcel, PixiJS and TypeScript. PixiJS started shipping with Type information since version 5, which is the one being included in the starter template. This means we don't have to import extra packages to be able to type-check PixiJS code.
+
+Note that the only real requirement for this is Node and NPM.
+
+### Project setup
+
+Let's go ahead and clone the template repo:
+
+```
+git clone --depth 1 https://github.com/llorenspujol/parcel-pixijs-quickstarter.git DinoGame
+```
+
+Immediately, on our **DinoGame** folder we'll have a starting project app that combines TypeScript and PixiJS (and already includes the type information), bundled with Parcel which also provides some nifty features such as hot module reloading. At this point it's a good idea to run `npm install` to install all the package dependencies that are defined as standard node dependencies.
+
+We can actually go ahead and run the application by doing `npm run start`, which kicks off the build process and hosts the app on `https://localhost:1234/`. You'll see a cute animation of a character that looks like Bomberman.
+
+Let's review the code a little bit. The two TypeScript files that are important are the following: `main.ts`, on the root directory, and `src\app.ts`
+
+`main.ts` is a simple file that imports from `app.ts` and creates a GameApp object by passing an HTML element, a number for screen width, and one for screen height. I want to make a low-res kind of game so I'll drop down the resolution to _300x75_:
+
+```ts
+import { GameApp } from "./app/app"
+const myGame = new GameApp(document.body, 300, 75)
+```
+
+Finally, we have `/assets/loader.js`, which is where the creator of the template left some code that loads the game assets. Let's modify it to suit our needs:
+
+```ts
+import ghost from "./images/ghost/*.png"
+import cloud from "./images/cloud/*.png"
+import obstacle1 from "./images/obstacle1/*.png"
+import obstacle2 from "./images/obstacle2/*.png"
```

Comment:
Maybe add a little comment about how `"./images/obstacle1/*.png"` and the entire import statement will import a series of images and that is why you are using `Object.values(obstacle1)`.

FYI the shape of `obstacle1` should be something like this

```
obstacle1 = {
  image1: pathToImage1, // or a data URI depending on how parcel is configured
  image2: pathToImage2,
}
// so when you Object.values
Object.values(obstacle1) = [
  pathToImage1, pathToImage2, ...
]
```

Reply:
Great point, thanks for the code illustration as well.